### PR TITLE
[ROCm] support for rocm5.3 wheel builds

### PR DESCRIPTION
**Pull request by jataylo**
-  Update common/install_miopen.sh to set build flags and staging branch for rocm5.3
-  Added dependencies for roc5.3 in build_rocm.sh
-  Updated bulild requirements for building magma

### CODE REVIEW

File: manywheel/build_common.sh [hunk 1]
```diff
@@ -162,11 +162,23 @@ else
...
+if [[ "$DISABLE_RCCL" = 1 ]]; then
+    echo "Disabling NCCL/RCCL in pyTorch"
+    USE_RCCL=0
+    USE_NCCL=0
+    USE_KINETO=0
+else
+    USE_RCCL=1
+    USE_NCCL=1
+    USE_KINETO=1
+fi
```
Comment (jithunnair-amd) [inline manywheel/build_common.sh:174]: @pruthvistony Should this patch be upstreamed?

Comment (pruthvistony) [reply manywheel/build_common.sh:174]: @jithunnair-amd ,
Recommend to keep this change in upstream, will be required to build pyTorch without NCCL/RCCL support and work through torch_UCC. On the code front, it is guarded by `DISABLE_RCCL` flag and should NOT affect any other builds.

**Review by pruthvistony — APPROVED** [review]

**Comment by jataylo** [comment]
@malfet could you take a look at this change for building rocm5.3 wheels, to set off any relevant CI we need

**Review by malfet — APPROVED** [review]

**Comment by jithunnair-amd** [comment]
Hi @seemethere/@malfet, the ROCm CI for this PR actually had failed due to a logic issue in the PR which also showed up in another later PR https://github.com/pytorch/builder/pull/1169#issuecomment-1287577830. I was wondering how we could have avoided this?

**Pull request merged**